### DISCUSSION TITLE
Tildes and diacritical marks

### DISCUSSION THREAD

**Discussion by Lobo-Feroz** [binarybottle/engram-es#5] [Ideas]
Extracted from the original discussion: https://github.com/binarybottle/engram-es/discussions/3

@binarybottle
How does one deal with diacritical marks for the seven additional letters?: Á, É, Í, Ñ, Ó, Ü and
(https://www.sttmedia.com/characterfrequency-spanish). Should these be accessed using key binding with some other key besides the Shift key, such as the Alt-Right key?

@Lobo-Feroz
The name in spanish for all of these diacritical marks is "tilde" (https://en.wikipedia.org/wiki/Tilde). Currently the writing for all of those is differentiated.

For simple tildes ÁÉÍÓÚ, as @NickG13 said, it's the key '´' followed by the vowel key.

The two dots is called "diaeresis" (https://en.wikipedia.org/wiki/Diaeresis_(diacritic)) and we have a key '¨' for that. Actually, it's the shift for the simple tilde key '´' (which, imho, it's not that bad). You can add these two dots to any vowel ÄËÏÖÜ äëïöü but it's only correct in the u. The rest are not used in proper spanish (but we can type them with our current keyboards).

The tilde for the Ñ is called "virgulilla" (weirdest name) and it's consireded a tilde as well. In qwerty-es we have a full key in the home row for the ñ, which I think it's very inefficient, as ñ has a frequency of 0.17% according to one of @NickG13 's links above. I think it could just be a modification on the n key: '´' + n = ñ

Now, here, I don't know what would be the best way to implement this tilde key. On qwerty-es, it's sequential, press tilde then press the vowel. An alternative could be simultaneous press with the opposite hand, but AltGr is already used for the vowel "e": AltGr + e = € (but just this one, 'aioun' don't have AltGr uses). Also, having a single key to switch to "tilde layer" would mean we'd need another method for the diaeresis '¨'.

**Comment by binarybottle** [comment]
This is my current plan for diacritical marks. Please tell me what you think:

ÁÉÍÓÚ Ñ -- Simultaneously press the corresponding letter key and the right-alt key.  

Ö -- Simultaneously press the colon key and the right-alt key.  

€ -- Simultaneously press the letter E key and the right-alt key.

**Reply by NickG13** [reply]
Hi! I've been experimenting with the alt key and when you keep pressing it for a long time it can become a bit painful for your thumbs. Since some letters with accent frecuency can get higher than some letters, it can become a problem in the long therm on high speeds. Maybe with an ergonomical physical layout that could be solved, or a wide mod like DreymaR from colemak community said once to me. 
Also, it seems that in ANSI you need to put Ctrl+Alt, so that would make it harder to write in ANSI.

**Reply by Lobo-Feroz** [reply]
I haven't tested it yet so I can't vouch for the comfort, but I can test it once we have a base layout. It's true what @NickG13 says that AltGr does not seem easy on the thumb.

You'll need a different combo for the eur € since "E + AltGr" would be É. 

I've thought a bit about this and maybe we could replace the $ with the €, and make the $ the AltGr version of the €. 
In Spain, we don't really use the dollar symbol that much. Although I suppose some latin american countries use the dollar more than Spain.
Another way would be to assing the $ to "S + AltGr", and/or using another char for the eur like "R + AltGr".

I forgot to mention that we could also do the "Cedilla" Ç with "C + AltGr". Ç is only used in the catalan language but I don't see any disadvantage on having it as an AltGr version. 

Another way would be to do that simultaneous strike for all these combos but keeping the tilde key ( ´ ) instead of using AltGr. Currently it's located at the left of the Ñ key, which means displacing the pinky one position to the right. Maybe (and depending on the actual frequencies of the tilded characters) it could be moved under the pinky, where the Ñ is right now, since we won't be having Ñ anymore. 

But I assume the 8 letters in the home row under the fingers will and should go to the most used characters. I don't know the data for the cumulative frequency of all tilded letters (the sum of ÁÉÍÓÚÜÑ), but I doubt it will be more than the 8 most used letters.

Another idea would be for the tilde to go in the place of the apostrophe in engram-en, that is center column upper row. Maybe in the right side instead of the left, to be alternate with the vowels. The apostrophe is very rare in spanish and I don't think it deserves a full main key to itself.

Also I think that the tilde is more used than the dot in spanish. As most sentences usually have one or more tilded vowels while only having one dot at the end.

I think that a nice next step would be to check all these assumptions by checking the frequencies of punctuation signs and see what are the actual values. @binarybottle can you point me to some of your code that I could use to check for punctuation frequencies?

**Reply by Lobo-Feroz** [reply]
For ease of reference, here's the qwerty-es keyboard.

As you can see, the Ñ (freq. 0.17%) is under the pinky. The tilde ( ´ ) key is the next one to the right.

Also, now that I'm looking at it, I must say we don't need at all neither the circumflex accent ( ^ ) https://en.wikipedia.org/wiki/Circumflex, nor the reverse tilde ( ` ) called Grave Accent https://en.wikipedia.org/wiki/Grave_accent . That key can and should be reused.

![qwerty-es](https://user-images.githubusercontent.com/10242383/127118495-7432c286-6051-4714-ad71-098e74edb52f.png)

**Comment by iandoug** [comment]
OK, my 2c :-)

1. US Ansi (and maybe Windows-specific) has 2 Alt keys. Euro boards and/or Linux have Alt on left and AltGr on right. I assume Windows also defaults to AltGr for non-English Euro languages.

2. There are two methods for diacritics ... dead keys or AltGr. AltGr is faster and uncomfortable in frequent use. Dead keys break the flow of typing, and sometimes need 2 keys to trigger plus the actual letter. The Germans use a dead key trigger on the Tilde key which is not user-friendly.
3. Note re terminology: tilde is ~, backtick is `, acute is ' (in terms of keys for diacritics)
4. KLA can give wrong scores if needed letters are not on the keyboard. Also it can only handle AltGr method, not deadkey method.

The real issue with languages that have more than 26 letters is the ANSI/ISO form factor ... it is just not suitable. Trying to make a Spanish or French or German layout with those limitations will end with suboptimal result.

Cheers, Ian

**Reply by Lobo-Feroz** [reply]
Hey @iandoug,

My 1c on top of your 2c would be that some keyboard layouts put the diacritics in the base keys. Like [AdNW](http://www.adnw.de/), ÜÄÖ are base keys. @NickG13 is doing the same with his AEIOU keyboard for ÁÉÍÓÚ. Of course, that means you have to move other characters to other less accessible locations.

>  The real issue with languages that have more than 26 letters is the ANSI/ISO form factor ... it is just not suitable. Trying to make a Spanish or French or German layout with those limitations will end with suboptimal result.

Yeah maybe an ergo ortholinear keyboard with the tilde under the right thumb would do the trick, or else we'd need a keyboard with 5-7 more keys to have a base key for each tilded character. But just thinking out loud. For this limitation I think we'll have to work with the available resources.

**Comment by binarybottle** [comment]
I would prefer a single, consistent method used across all diacriticals, whether it be a preceding key or a simultaneous key. What I'm hearing from your comments is it would be good to have a dedicated preceding key rather than a simultaneously pressed key. Is that correct?

**Comment by binarybottle** [comment]
Please see https://github.com/binarybottle/engram-es/discussions/7#discussioncomment-1100537
